### DISCUSSION TITLE
Vuforia - reconhecimento de imagem por API

### DISCUSSION THREAD

**Discussion by andreluiztorres** [Flutterando/forum#81] [Perguntas e Respostas]
SOCORRO ... no app que estou desenvolvendo, existe um módulo que apos tirar foto deve reconhecer uma imagem hospedada em Vuforia e retornar um ID. (Vuforia Query Web API). 

consegui chegar até este ponto:

_vuforia() async {
    const boundary = "----------ThIs_Is_tHe_bouNdaRY_\$";
    const contentType = "multipart/form-data; boundary=$boundary";
    final imageBinary = await image!.readAsBytes();
    // ignore: prefer_interpolation_to_compose_strings
    var body = "--" +
        boundary +
        "\r\n" +
        'Content-Disposition: form-data; name="image"; filename="image.jpg"\r\n' +
        "Content-Type: image/jpeg\r\n\r\n" +
        imageBinary.toString() +
        "\r\n" +
        "--" +
        boundary +
        "--\r\n";
    var content = utf8.encode(body);
    const md5 = crypto.md5;
    var digest = md5.convert(content);
    var contentMd5 = base64.encode(digest.bytes);
    var verb = 'POST';
    var type = 'multipart/form-data';
    var date = HttpDate.format(DateTime.now());
    var path = '/v1/query';
    var stringToSign = '$verb\n$digest\n$type\n$date\n$path';
    var signature = base64.encode(Hmac(sha1, global.clientSecretKey.codeUnits)
        .convert(stringToSign.codeUnits)
        .bytes);
    var authorization = 'VWS ${global.clientAccessKey}:$signature';

    final response = await http.post(
      Uri.parse("https://cloudreco.vuforia.com/v1/query"),
      headers: {
        'Content-Type': contentType,
        'Authorization': authorization,
        'Date': date,
        'Content-MD5': contentMd5,
      },
      body: body,
    );

    print(response.body);

...

Consigo receber resposta da API atuenticada, porem retorna BadImage.. ou seja .. certamente o formato que esotu enviando a imageem está no formato erado.

para chegar nesse resultado me basiei num outro fonte em NODE.JS que testei e funciona, porém a imagem é enviada usando:

fs.readFileSync(filename, 'binary')

Tentei enviar convertendo a imagem através de: 

final imageBinary = await image!.readAsBytes();

mas sem sucesso.

quero muito finalizar este processo e publicar no Pub.Dev para poder ajudar a comunidade. A ferramenta é muito bacana, mas quase não existe documentação explicando como integrar ao flutter.

Muito obrigado desde já a todos.

**Comment by renanzdm** [comment]
Pode ser que você esteja tentando enviar de forma errada, aqui tenho um método que uso para fazer upload de imagens

```
Future<HttpResponse> uploadFile(
      {required File file, required String url}) async {
    var request = MultipartRequest(
      'POST',
      Uri.parse('$baseurl$url'),
    );
    Map<String, String> headers = {"Content-type": "multipart/form-data"};
    request.files.add(
      await MultipartFile.fromPath(
        'file',
        file.path,
      ),
    );
    request.headers.addAll(headers);
    StreamedResponse streamedResponse = await request.send();
    Response response = await Response.fromStream(streamedResponse);
    return HttpResponse(
        statusCode: response.statusCode, data: jsonDecode(response.body));
  }
}```

**Reply by andreluiztorres** [reply]
@renanzdm .. valew pwla dica .. estou tentando integrar a meu codigo sua dica .. mas de uma duas .. ou não funcionou por que não era a resposta, ou não consegui integrar sua ideia a meu fonte (o que acho muito mais factivel kkkk)

**Comment by renanzdm** [comment]
Se seu codigo for público eu poderia dar uma olhada assim que possível

**Reply by andreluiztorres** [reply]
Sería esse código que coloquei a cima mesmo... só precisaria de novas cresenciais para teste
